profit/util/variable_kinds.py

```diff
@@ -30,9 +33,329 @@ def independent(start=0, end=1, step=1, size=None):
+        """Generates a common halton sequence for all variables where this is possible and transforms them
+        according to their distribution."""
+        halton_variables = [v for v in self.list if v.kind.lower() not in EXCLUDE_FROM_HALTON]
+        if halton_variables:
+            nd_halton_seq = halton((self.samples, len(halton_variables)))
+            for idx, v in enumerate(halton_variables):
+                v.generate_values(nd_halton_seq[:, idx])
+
+    def resolve_dependent(self):
+        """Create a Variable instance for the independent variables of vector outputs."""
+        output = [v for v in self.list if 'output' in v.kind.lower() and v.dependent]
+        for o in output:
+            dvars = []
+            for d in o.dependent:
+                dv = [v for v in self.list if v.name == d][0]
+                o.size = (o.size[0], dv.value.shape[0])
+                o.value = np.full(o.size, np.nan)
+                dvars.append(dv)
+            o.dependent = dvars
+
+
+class Variable:
+    """Class for a single variable.
+
+    Attributes:
+        name (str): Name of the variable.
+        kind (str): Distribution for input variables, 'Output' or 'Independent'.
+        size (tuple): Size as (nsamples, ndim).
+        value (ndarray): Data.
+        dtype (dtype): Datatype.
+    """
+
+    def __init__(self):
+        self.name = ''
+        self.kind = ''
+        self.size = (1, 1)
+        self.value = np.array([[]])
+        self.dtype = np.float64
+
+    @property
+    def named_value(self):
+        """
+        Returns:
+            Ndarray with dtype.
+        """
+        return np.array(self.value, dtype=[(self.name, self.dtype)])
+
+    @classmethod
+    def create_from_str(cls, name, size, v_str):
+        """Creates a Variable instance from a string. E.g. 'Uniform(3.4, 7.8)'
+
+        Parameters:
+            name (str): Name of the variable.
+            size (tuple): Size as (nsamples, ndim).
+            v_str (str): String from which the variable is constructed.
+
+        Returns:
+            Variable.
+        """
+        from re import split
+
+        def try_parse(s):
+            funcs = [int, float]
+            for f in funcs:
+                try:
+                    return f(s)
+                except ValueError:
+                    pass
+            return s
+
+        parsed = split('[()]', v_str)
+        kind = parsed[0]
+        args = parsed[1] if len(parsed) >= 2 else ''
+        entries = tuple(try_parse(a) for a in args.split(',')) if args != '' else tuple()
+
+        if isinstance(try_parse(v_str), (int, float)):  # PyYaml has problems parsing scientific notation
+            kind = 'Constant'
+            entries = (try_parse(v_str),)
+
+        v_dict = {'name': name, 'kind': kind, 'size': size, 'entries': entries}
+        return cls.create(**v_dict)
+
+    @classmethod
+    def create(cls, name, kind, size, entries=(0, 1), value=np.array([[]]), dtype=np.float64):
```

Comment:
Using `np.array([[]])` as a default value is dangerous, as the same value would be shared across all `Variable` objects. (similar to how a `list` as a default value works) In this case it might just work anyways.

Also, why do you use `create` instead of just `__init__`? I'm just curious.